### PR TITLE
[JSDoc] Add missing documentation for botbuilder/skills

### PR DESCRIPTION
**Pull request by ceciliaavila**
Addresses # 2602

## Description
This PR adds the missing documentation in [botbuilder](https://github.com/microsoft/botbuilder-js/tree/master/libraries/botbuilder) library.
Some documentation is ported and adapted from [botbuilder-dotnet](https://github.com/microsoft/botbuilder-dotnet)

_Note: We divided the work for this library into two PRs to easy their review. The PR# 144 contains the rest of the botbuilder fixes._

## Specific Changes

- Added JSDoc comments in all public methods and some of the private ones in the following files:
    - activityValidator
    - botFrameworkAdapter
    - botFrameworkHttpClient
    - channelServiceHandler
    - channelServiceRoutes
    - skills/skillHandler
    - skills/skillHttpClient
    - streaming/streamingHttpClient
    - streaming/tokenResolver

## Testing
The following image shows the ESLint Report with no JSDoc errors.
image

### CODE REVIEW

File: libraries/botbuilder/src/botFrameworkAdapter.ts [hunk 1]
```diff
@@ -544,6 +550,15 @@ export class BotFrameworkAdapter extends BotAdapter implements ConnectorClientBu
...
     public async getUserToken(context: TurnContext, connectionName: string, magicCode?: string): Promise<TokenResponse>;
     public async getUserToken(context: TurnContext, connectionName: string, magicCode?: string, oAuthAppCredentials?: CoreAppCredentials): Promise<TokenResponse>;
+    /**
+     * Asynchronously attempts to retrieve the token for a user that's in a login flow.
+     * @param context The context object for the turn.
+     * @param connectionName The name of the auth connection to use.
+     * @param magicCode Optional. The validation code the user entered.
+     * @param oAuthAppCredentials Optional. AppCredentials for OAuth.
+     * 
+     * @returns A [TokenResponse](xref:botframework-schema.TokenResponse) object that contains the user token.
```
Comment (santgr11) [inline libraries/botbuilder/src/botFrameworkAdapter.ts:560]: Wen clicking on the xref in vs code I get an error. Is this the proper way to do this?
image

Also, I think we are not using a space between the params and returns.

**Review by sw-joelmut — APPROVED** [review]

File: libraries/botbuilder/src/botFrameworkAdapter.ts (hunk 1 above)
Comment (ceciliaavila) [reply libraries/botbuilder/src/botFrameworkAdapter.ts:560]: In this case, I'd keep it like this, as it's a method overload and it has the almost exact documentation that the methods above.

**Review by santgr11 — APPROVED** [review]

**Review by iscaltritti-sw — CHANGES_REQUESTED** [review]

File: libraries/botbuilder/src/botFrameworkHttpClient.ts [hunk 2]
```diff
@@ -56,6 +61,18 @@ export class BotFrameworkHttpClient implements BotFrameworkClient {
      */
     public async postActivity<T>(fromBotId: string, toBotId: string, toUrl: string, serviceUrl: string, conversationId: string, activity: Activity): Promise<InvokeResponse<T>>
     public async postActivity(fromBotId: string, toBotId: string, toUrl: string, serviceUrl: string, conversationId: string, activity: Activity): Promise<InvokeResponse>
+    /**
+     * Forwards an activity to another bot.
+     * @remarks
```
[the remark this answers is not in the document]
Comment (ceciliaavila) [reply libraries/botbuilder/src/botFrameworkHttpClient.ts]: Applied. Thanks!

**Review by iscaltritti-sw — APPROVED** [review]

**Review by denscollo — APPROVED** [review]

**Comment by denscollo** [comment]
Promoted

**Pull request closed without merging**